docs/source/command_line.rst

```diff
@@ -360,11 +360,21 @@ Here are some more useful flags:
   updates the cache, but regular incremental mode ignores cache files
   written by quick mode.
 
+- ``--python-executable EXECUTABLE``  This flag will attempt to set
+  ``--python-version`` if not already set based on the interpreter given.
+
 - ``--python-version X.Y`` will make mypy typecheck your code as if it were
   run under Python version X.Y. Without this option, mypy will default to using
   whatever version of Python is running mypy. Note that the ``-2`` and
   ``--py2`` flags are aliases for ``--python-version 2.7``. See
-  :ref:`version_and_platform_checks` for more about this feature.
+  :ref:`version_and_platform_checks` for more about this feature. This flag
+  will attempt to find a Python executable of the corresponding version. If
+  you'd like to disable this, see ``--no-infer-executable`` below.
+
+- ``--no-infer-executable`` will disable searching for a usable Python
```

Comment:
I don't think this flag makes any sense without PEP561, and that `--no-site-packages` was a better name in that scenario, as it indicated the purpose of the executable

Reply:
Hm, I think I might agree in principle with the first (but I split this out for ease of review mostly). 

As for the flag name, I'm not so sure. If the executable is ever used for anything other than PEP 561, the `--no-site-packages` flag would be misleading. It is misleading now anyway, it doesn't actually disable searching, it sets `options.python_executable` to `None`, which happens to mean that searching for PEP 561 packages is not done.

Reply:
> it sets options.python_executable to None, which happens to mean that searching for PEP 561 packages is not done.

You've got this backwards. `options.python_executable is None` is an implementation-detail of how we represent "don't do pep561 searching", rather than storing a redundant boolean alongside it. We could add a `@property` to `Options` to make that more explicit, I suppose.

We shouldn't be exposing that detail in the command line arguments.

Reply:
Hm, I suppose from a user perspective it would be better to name it `--no-site-packages`.